lib/from_openstudio/construction/shade.rb

```diff
@@ -0,0 +1,74 @@
+# *******************************************************************************
+# Honeybee OpenStudio Gem, Copyright (c) 2020, Alliance for Sustainable
+# Energy, LLC, Ladybug Tools LLC and other contributors. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# (1) Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# (2) Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# (3) Neither the name of the copyright holder nor the names of any contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission from the respective party.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'honeybee/construction/shade'
+
+require 'from_openstudio/model_object'
+
+module Honeybee
+  class ShadeConstruction < ModelObject
+
+    def self.from_construction(construction)
+        # create an empty hash
+        hash = {}
+        hash[:type] = 'ShadeConstruction'
+        # set hash values from OpenStudio Object
+        hash[:identifier] = construction.nameString
+        hash[:materials] = []
+        # get construction layers
+        layers = construction.layers
+        i = 0
+        layers.each do |layer|
```

Comment:
I am unclear on why you are looping through the layers here. We only need to check the outer-most layer and all of the other layers that you are looping through have no effect on the resulting shade construction.  The other checks you have here about glazing vs. opaque materials looks good.  However, I think NoMass materials will also be a fairly common way that people specify the reflectance/absorptance of shades so we should also check for those.

Reply:
In this case, I looped through all layers and am getting only the outermost, since we were looping through them anyway to get the namestring for each layer.

Reply:
The NoMass materials do not have a reflectance property in OS.

https://openstudio-sdk-documentation.s3.amazonaws.com/cpp/OpenStudio-3.1.0-doc/model/html/classopenstudio_1_1model_1_1_massless_opaque_material.html

Reply:
But `NoMass` material layers have a `solarAbsorptance`, which is always equal to `1 - solar reflectance`.  If you want to do a test to see if `NoMass` materials can't be used to Shades, then we can forgo the check. Otherwise, I'm inclined to think that they are permissible.